Build failure: framework-laptop-kmod with kernel 6.11-rc7 aka testing

### Steps To Reproduce

Steps to reproduce the behavior:
1. build framework-laptop-kmod with linuxPackages_testing

### Build log

```
error: builder for '/nix/store/i0x14fn2zrbq0c3awcy515n9gi62wgl3-framework-laptop-kmod-0-unstable-2024-01-02.drv' failed with exit code 2;
       last 10 log lines:
       > /build/source/framework_laptop.c:358:19: error: initialization of 'void (*)(struct platform_device *)' from incompatible pointer type 'int (*)(struct platform_device *)' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Werror=incompatible-pointer-types8;;]
       >   358 |         .remove = framework_remove,
       >       |                   ^~~~~~~~~~~~~~~~
       > /build/source/framework_laptop.c:358:19: note: (near initialization for 'framework_driver.<anonymous>.remove')
       > cc1: some warnings being treated as errors
       > make[3]: *** [/nix/store/cbbxx7s9b116h9yrqasmizwn62cl6vz7-linux-6.11-rc7-dev/lib/modules/6.11.0-rc7/source/scripts/Makefile.build:244: /build/source/framework_laptop.o] Error 1
       > make[2]: *** [/nix/store/cbbxx7s9b116h9yrqasmizwn62cl6vz7-linux-6.11-rc7-dev/lib/modules/6.11.0-rc7/source/Makefile:1926: /build/source] Error 2
       > make[1]: *** [/nix/store/cbbxx7s9b116h9yrqasmizwn62cl6vz7-linux-6.11-rc7-dev/lib/modules/6.11.0-rc7/source/Makefile:224: __sub-make] Error 2
       > make[1]: Leaving directory '/nix/store/cbbxx7s9b116h9yrqasmizwn62cl6vz7-linux-6.11-rc7-dev/lib/modules/6.11.0-rc7/build'
       > make: *** [Makefile:12: modules] Error 2
       For full logs, run 'nix log /nix/store/i0x14fn2zrbq0c3awcy515n9gi62wgl3-framework-laptop-kmod-0-unstable-2024-01-02.drv'.
error: 1 dependencies of derivation '/nix/store/0wh4xcbmknsi4nf71wzdls2va8ki39nx-linux-6.11-rc7-modules.drv' failed to build
error: 1 dependencies of derivation '/nix/store/fy93qhi4dr2kdyxg7y992ldgkz091y1r-nixos-system-framework-24.05.20240914.8f7492c.drv' failed to build
```

### Additional context

Wanted to try the 6.11 Kernel cause there is a bug fixed in 'cros_ec' module which pings in kde wrong battery state

https://community.frame.work/t/responded-battery-percentage-gets-misreported-as-0-occasionally/38056/9

### Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.10.10, NixOS, 24.05 (Uakari), 24.05.20240914.8f7492c`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.5`
 - channels(root): `"home-manager-24.05.tar.gz, nixos-24.05, nixos-hardware, nixos-unstable"`
 - nixpkgs: `/nix/store/s8pb2vy9zdjxlc5gr2iwkkm9y5gqr4ph-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/os-specific/linux/framework-laptop-kmod/default.nix
+++ b/pkgs/os-specific/linux/framework-laptop-kmod/default.nix
@@ -1,19 +1,19 @@
 { lib
 , stdenv
-, linuxPackages
 , kernel
 , fetchFromGitHub
+, unstableGitUpdater
 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation {
   pname = "framework-laptop-kmod";
-  version = "0-unstable-2024-01-02";
+  version = "0-unstable-2024-09-15";
 
   src = fetchFromGitHub {
     owner = "DHowett";
     repo = "framework-laptop-kmod";
-    rev = "a9e8db9ba2959b75c1fb820ffac8fa189f0f63c3";
-    hash = "sha256-Ai/OxvkaKPltri8R0oyfmxQLUVfaj6Q8vebrhmWYhUU=";
+    rev = "6164bc3dec24b6bb2806eedd269df6a170bcc930";
+    hash = "sha256-OwtXQR0H4GNlYjVZ5UU5MEM6ZOjlV3B0x2auYawbS2U=";
   };
 
   nativeBuildInputs = kernel.moduleBuildDependencies;
@@ -28,11 +28,15 @@ stdenv.mkDerivation rec {
     runHook postInstall
   '';
 
+
+  passthru.updateScript = unstableGitUpdater { };
+
   meta = with lib; {
     description = "Kernel module that exposes the Framework Laptop (13, 16)'s battery charge limit and LEDs to userspace";
     homepage = "https://github.com/DHowett/framework-laptop-kmod";
     license = licenses.gpl2Only;
     maintainers = with maintainers; [ gaykitty ];
     platforms = platforms.linux;
+    broken = lib.versionOlder kernel.version "6.1";
   };
 }

